brush/src/hooks/useWhenVisionChanges.js

```diff
@@ -0,0 +1,10 @@
+import { useContext, useEffect } from 'react'
+import VisionContext from '../context/VisionContext'
+
+const useFuncWhenVisionChanges = (callback) => {
```

Comment:
This name could be simplier like his
```suggestion
const useWhenVisionChanges = (callback) => {
```